### ISSUE TITLE
PAK Soundscapes and PAK files can't be located.

### ISSUE THREAD

**Issue by Nak2** [Facepunch/garrysmod-issues#5059]
### Details
sound.GetTable and sound.GetProperties seems to ignore all custom soundscripts within a map's PAK.

I've also been trying to locate the soundscape by other means, but file.Find ignores all PAK files as well.
It is only file.Exists that returns true.

### Steps to reproduce
Load a map with custom soundscape script ( In this example I'm using https://steamcommunity.com/sharedfiles/filedetails/?id=2028581413 )

Trying to locate the file using file.Find doesn't return any results:
`lua_run PrintTable((file.Find("scripts/soundscapes_rp*", "GAME")))`  -> Empty Table

file.Exists confirm it is a valid file (Can also read it too):
`lua_run print(file.Exists("scripts/soundscapes_rp_southside_day.txt", "GAME"))` -> true

Materials and other PAK content are also unsearchable:
`lua_run PrintTable((file.Find("materials/bc/*", "GAME")))` -> Empty Table

Trying to use sound.GetProperties or sound.GetTable will also fail to list the custom sounds:
`lua_run print(sound.GetProperties("SS.Streets"))` -> Empty string

### TLDR
`file.Find` `sound.GetTable` and `sound.GetProperties` ignores all files within a maps PAK.

**Comment by robotboy655** [comment]
Looks like searching through map file PAK lump was never implemented.

Also soundscapes are not sound**scripts** which is why they won't appear in `sound.GetTable`/`Properties`.

**Comment by Nak2** [comment]
> 
> 
> Also soundscapes are not sound**scripts** which is why they won't appear in `sound.GetTable`/`Properties`.
Damn, I thought it worked like that. Guess my only option is to build my own system from scratch like I did before.

> Looks like searching through map file PAK lump was never implemented.
Are you able to make the few files in PAK lump searchable? I can provide the LUA code I've made so far to search filenames within zipfiles if you want (Does take ~0.5 seconds for 800MB PAK-lump).
I also think making a "PAK" path will make the community happy. Alternative rename it to "MAP" or "MapFiles".

**Comment by robotboy655** [comment]
The search path ID for map only is `BSP`.

I have made `file.Find` work with files embedded into the map. For now it works with both `GAME` and `BSP`. I have fears of it slowing down filesystem search across the board, or breaking something, so I might have to limit search to `BSP` only in that case.
